integration/tests/protect-jwt-v2.test.ts

```diff
@@ -0,0 +1,188 @@
+import type { OrganizationMembershipRole } from '@clerk/backend';
+import { expect, test } from '@playwright/test';
+
+import { appConfigs } from '../presets';
+import type { FakeOrganization, FakeUser } from '../testUtils';
+import { createTestUtils, testAgainstRunningApps } from '../testUtils';
+
+testAgainstRunningApps({
+  withEnv: [appConfigs.envs.withBillingStaging],
+})('has() for JWT v2 @nextjs', ({ app }) => {
+  test.describe.configure({ mode: 'serial' });
+
+  let fakeAdmin: FakeUser;
+  let fakeReader: FakeUser;
+  let fakeManager: FakeUser;
+  let fakeOrganization: FakeOrganization;
+
+  test.beforeAll(async () => {
+    const m = createTestUtils({ app });
+    fakeAdmin = m.services.users.createFakeUser();
+    const admin = await m.services.users.createBapiUser(fakeAdmin);
+    fakeOrganization = await m.services.users.createFakeOrganization(admin.id);
+    fakeReader = m.services.users.createFakeUser();
+    const reader = await m.services.users.createBapiUser(fakeReader);
+    await m.services.clerk.organizations.createOrganizationMembership({
+      organizationId: fakeOrganization.organization.id,
+      role: 'org:impersonation_reader' as OrganizationMembershipRole,
+      userId: reader.id,
+    });
+    fakeManager = m.services.users.createFakeUser();
+    const manager = await m.services.users.createBapiUser(fakeManager);
+    await m.services.clerk.organizations.createOrganizationMembership({
+      organizationId: fakeOrganization.organization.id,
+      role: 'org:manager' as OrganizationMembershipRole,
+      userId: manager.id,
+    });
+  });
+
+  test.afterAll(async () => {
+    await fakeOrganization.delete();
+    await fakeManager.deleteIfExists();
+    await fakeAdmin.deleteIfExists();
+    await fakeReader.deleteIfExists();
+    await app.teardown();
+  });
+
+  test('`has()` in RSCs and RCCs as `admin`', async ({ page, context }) => {
+    const u = createTestUtils({ app, page, context });
+
+    await u.po.signIn.goTo();
+    await u.po.signIn.waitForMounted();
+    await u.po.signIn.signInWithEmailAndInstantPassword({ email: fakeAdmin.email, password: fakeAdmin.password });
+    await u.po.expect.toBeSignedIn();
+
+    await u.po.organizationSwitcher.goTo();
+    await u.po.organizationSwitcher.waitForMounted();
+    await u.po.organizationSwitcher.waitForAnOrganizationToSelected();
+
+    await u.page.goToRelative('/jwt-v2-organizations');
+
+    await expect(u.page.getByTestId('client-jwt').getByText(`"per": "create,delete,manage,read"`)).toBeVisible();
```

Comment:
Instead of checking that a given substring is present on the page, did you consider getting the `textContent` of the selector and running that through `JSON.parse` so you can directly compare values?